pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -304,24 +304,36 @@ func (r *ReconcileApplicationLayer) Reconcile(ctx context.Context, request recon
 
 // updateApplicationLayerWithDefaults populates the applicationlayer with defaults.
 func updateApplicationLayerWithDefaults(al *operatorv1.ApplicationLayer) {
-	defaultLogIntervalSeconds := int64(5)
-	defaultLogRequestsPerInterval := int64(-1)
+	var defaultLogIntervalSeconds int64 = 5
+	var defaultLogRequestsPerInterval int64 = -1
+	var defaultLogCollectionDisabled operatorv1.LogCollectionStatusType = operatorv1.L7LogCollectionDisabled
+	var defaultWebApplicationFirewallDisabled operatorv1.WAFStatusType = operatorv1.WAFDisabled
 
-	if al.Spec.LogCollection != nil {
+	if al.Spec.LogCollection == nil {
+		al.Spec.LogCollection = new(operatorv1.LogCollectionSpec)
+		al.Spec.LogCollection.CollectLogs = &defaultLogCollectionDisabled
+	}
+
+	if *al.Spec.LogCollection.CollectLogs == operatorv1.L7LogCollectionEnabled {
 		if al.Spec.LogCollection.LogRequestsPerInterval == nil {
 			al.Spec.LogCollection.LogRequestsPerInterval = &defaultLogRequestsPerInterval
 		}
 		if al.Spec.LogCollection.LogIntervalSeconds == nil {
 			al.Spec.LogCollection.LogIntervalSeconds = &defaultLogIntervalSeconds
 		}
 	}
+
+	if al.Spec.WebApplicationFirewall == nil {
+		al.Spec.WebApplicationFirewall = &defaultWebApplicationFirewallDisabled
+	}
 }
 
 // validateApplicationLayer validates ApplicationLayer
 func validateApplicationLayer(al *operatorv1.ApplicationLayer) error {
 
 	// If ApplicationLayer spec exists then one of its features should be set.
-	if al.Spec.LogCollection == nil && al.Spec.WebApplicationFirewall == nil {
+	if *al.Spec.LogCollection.CollectLogs != operatorv1.L7LogCollectionEnabled &&
```

Comment:
We should still take care to check pointer dereferences here, as a safety precaution - we don't want to panic ever.

Reply:
Sure, I will add it here also.